documentation/smg/16_serialiser.md

```diff
@@ -0,0 +1,229 @@
+
+__NB.__ For struct arrays `struct2cell` produces a Rank-(N+1) cell array, this means that for struct arrays, the limit on rank is 6.
+
+#### Cell array
+Cell array data (type tag 23) are serialised as:
+
+_Rank_ |     tag    | dim1 | dim2 | ... | Data
+-------|------------|------|------|:---:|------
+_NULL_ | 32 + type  | 0
+_0_    |   type     |      |      |     | Data
+_1_    | 32 + type  | nElem|      |     | Data
+_2_    | 64 + type  | dim1 | dim2 |     | Data
+_N_    | N<<5 + type| dim1 | dim2 | ... | Data
+
+Where `data` is the concatenation of the serialisation of each element.
+
+#### Object Array
+Object array data (type tag 26) are serialised as:
+
+_Rank_ |     tag    | dim1 | dim2 | ... |Data|     ...            |    ...     |   ...   |  ...
+-------|------------|------|------|:---:|----|--------------------|------------|---------|------
+_NULL_ | 32 + type  | 0    |      |     | 33 | length(class_name) | class_name |         |
+_0_    |    type    |      |      |     | 33 | length(class_name) | class_name | ser_tag | Data |
+_1_    | 32 + type  | nElem|      |     | 33 | length(class_name) | class_name | ser_tag | Data |
+_2_    | 64 + type  | dim1 | dim2 |     | 33 | length(class_name) | class_name | ser_tag | Data |
+_N_    | N<<5 + type| dim1 | dim2 | ... | 33 | length(class_name) | class_name | ser_tag | Data |
+
+Where the first 3 elements of data 
+(`33`,`length(class_name)`,`class_name`) are a serialised representation of the class name (see [Simple data][simpledata]),
+`ser_tag` is a `uint8` tag (described below) and
+`data` is the serialisation of the object by means described by `ser_tag`
+
+###### ser_tag
+The object array's `ser_tag` contains information about the means by which the object has been serialised.
+It can have one of three values:
+
+0) The object has been serialised by its own `serialize` method
+1) The object has been serialised by the `saveobj` method or function
+2) The object has been serialised by calling `struct` on the object
+
+#### Function handles
+Function handles are serialised differently depending on their type.
+As function handles in MATLAB are purely scalar objects, the rank-component of the tag is utilised for tagging the type of function handle (thus the method to [de]serialise), these types are:
+
+Handle type         |   tag
+--------------------|--------
+Simple/Class simple | 32 + 25
+Anonymous           | 64 + 25
+Scoped              | 96 + 25
+
+Function handles are serialised as:
+Handle type         |   tag   | data                                       | ...
+--------------------|---------|--------------------------------------------|-----------------------------------------
+Simple/Class simple |  57     | function name as serialised string         |
+Anonymous           |  89     | anonymous function as serialised string    | relevant workspace as serialised struct
+Scoped              |  121    | function parentage as serialised cell array|
```

Comment:
Again it's the data structure. If you are going to include this, it would be worth including what 'relevant workspace' means. I note that matlab documents function workspaces, but it is not clear that a function would need its function workspace serialising unless you are serialising in the middle of the function call. Is that the case here?

Reply:
From what I understand if you have an anonymous function which calls upon static variables defined in their workspace, this becomes their relevant workspace. So if you have an object which has a function as part of its operation. E.g.
```a = PixData(...); 
n = 1.5;
b = @(x)(x * n); % Trying to increase contrast
a = apply(b, a); % I don't know what the actual Horace syntax will be
```
If we want to apply `b` in parallel, all processes need to know what `b` is, and `b` depends on `n` (trivially in this case, but not in general), hence `n` would be part of `b`'s relevant workspace.

Reply:
@oerc0122 So it's the workspace for the anonymous function option value 89, not a general datum for all the options. If that were specified in the "relevant workspace" section this would be fine.

Reply:
@oerc0122 And sorry, looking at the format and realising you actually intend this, as it's aligned with 89, but it looks to a casual reader as if it applies to all the options. May still be worth specifying to avoid this.

Reply:
Don't forget in the rendered document this is a table with a blank space there. 

https://github.com/pace-neutrons/Horace/blob/233_c%2B%2B_serialiser_smg/documentation/smg/16_serialiser.md#function-handles